### PR TITLE
Fixes and enhancements regarding PPConfigManager and PP_CONFIG_PATH

### PR DESCRIPTION
**Pull request by TrogloGeek**

### CODE REVIEW

File: lib/PayPal/Core/PPConfigManager.php [hunk 1]
```diff
@@ -44,11 +56,13 @@ public static function getInstance()
 
 	//used to load the file
 	private function load($fileName) {
-
         //Gracefully check for ini file
-        if(@parse_ini_file($fileName)) {
-            $this->config = @parse_ini_file($fileName);
-        }
+		$parsedConfig = @parse_ini_file($fileName);
```
Comment (jaypatel512) [inline lib/PayPal/Core/PPConfigManager.php]: Minor Formatting Issue here. Can you please fix that ?

File: lib/PayPal/Core/PPConfigManager.php [hunk 2]
```diff
@@ -23,14 +23,26 @@ class PPConfigManager {
...
 	private static $instance;
 
-	private function __construct(){
-		if(defined('PP_CONFIG_PATH')) {
-			$configFile = constant('PP_CONFIG_PATH') . '/sdk_config.ini';
+	private function __construct() {
+		$configFile = null;
+		if (defined('PP_CONFIG_PATH')) {
+			if (strlen(PP_CONFIG_PATH) > 0) {
+				//let's allow setting full file path instead of containing folder,
```
Comment (jaypatel512) [inline lib/PayPal/Core/PPConfigManager.php]: 👍

File: lib/PayPal/Core/PPConfigManager.php [hunk 3]
```diff
@@ -23,14 +23,26 @@ class PPConfigManager {
...
+					// legacy behavior
+					$configFile = PP_CONFIG_PATH . DIRECTORY_SEPARATOR . 'sdk_config.ini';
+				}
+			}
 		} else {		
 			$configFile = implode(DIRECTORY_SEPARATOR,
 				array(dirname(__FILE__), "..", "config", "sdk_config.ini"));
 		}
-		$this->load($configFile);
+		if (!is_null($configFile)) {
```
Comment (jaypatel512) [inline lib/PayPal/Core/PPConfigManager.php]: Can we also add the logic to check if `file_exists` as shown here: https://github.com/paypal/PayPal-PHP-SDK/blob/master/lib/PayPal/Core/PayPalConfigManager.php#L42

**Comment by jaypatel512** [comment]
Hey @TrogloGeek !

Thanks for the PR. This looks good. We have a similar ConfigManager in [PayPal-PHP-SDK](https://github.com/paypal/PayPal-PHP-SDK/blob/master/lib/PayPal/Core/PayPalConfigManager.php) that we moved from here when we removed the dependency. I have added some tiny improvements that would make this a great PR. 

Can you please take a look at it pretty quick ?

**Comment by TrogloGeek** [comment]
Hi, sorry for the latency, GMT+2/CEST here ;).

File: lib/PayPal/Core/PPConfigManager.php (hunk 1 above)
Comment (TrogloGeek) [inline lib/PayPal/Core/PPConfigManager.php]: Sorry, PSR-2 config here and did not notice. I'll fix it.

File: lib/PayPal/Core/PPConfigManager.php (hunk 3 above)
Comment (TrogloGeek) [inline lib/PayPal/Core/PPConfigManager.php]: I wanted to, but I thought maybe you chose to drastically limit disk IO. I'll add a check with great pleasure.

**Comment by TrogloGeek** [comment]
The ```is_file(PP_CONFIG_PATH)``` was a bad move, didn't thought this correctly, this constant is used as a config directory and sdk_config.ini is not the only file that is fetched using it, ```PPCertificateCredential``` also uses it to fetch the certificate when it is constructed with ```$certPath``` parameter not being an absolute path.

I'll revert this as it now seems to me a really bad idea, handy in my case but creating too much alternative use cases with ```PP_CONFIG_PATH```.
I'll keep the case where ```PP_CONFIG_PATH``` is defined empty, as the only interpretation I can see of this case (if not an error) is the person wants to disable loading files from this directory.

**Comment by TrogloGeek** [comment]
@jaypatel512 I rebased the pull request according to comments. Let me know.

**Comment by jaypatel512** [comment]
Looks good ! Let me run some quick tests, and get back to you. 👍  for getting back on this really quickly. Appreciated

**Comment by TrogloGeek** [comment]
Sorry, I forgot to edit the last commit message when I reverted the is_file(PP_CONFIG_PATH) logic, I just fixed it.

**Comment by jaypatel512** [comment]
👍 

Thanks @TrogloGeek for amazing work.

**Comment by TrogloGeek** [comment]
Thanks for merging this, not much work here but I badly need this fix (PHP 7 project, Issue 70 generates Fatal Error under PHP 7) and did not want to introduce such technical debt as working with a fork of the SDK.
When you want your PR to be merged fast, you make it right and you think about other use cases than yours ;-).

**Pull request merged**